core/src/main/java/hudson/cli/CommandDuringBuild.java

```diff
@@ -45,7 +45,7 @@
     protected Run getCurrentlyBuilding() throws CmdLineException {
         Run r = optCurrentlyBuilding();
         if (r==null)
-            throw new CmdLineException("This CLI command works only when invoked from inside a build");
+            throw new IllegalStateException("This CLI command works only when invoked from inside a build");
```

Comment:
In my opinion no, the issue isn't an incorrect number (or invalid content) of cmdline but the state of Job/Jenkins etc. In that case I'd prefer to not show an usage help msg.